AnkiDroid/src/test/java/com/ichi2/anki/ReviewerTest.kt

```diff
@@ -66,6 +69,44 @@ class ReviewerTest : RobolectricTest() {
         }
     }
 
+    @Test
+    fun testOnSelectedTags() {
+        // Add a note using basic model
+        addNoteUsingBasicModel()
+
+        // Start the Reviewer activity
+        val viewer = startRegularActivity<Reviewer>()
+
+        // Create a list of tags
+        val tags = listOf("tag1", "tag2")
+
+        // Define an arbitrary filter
+        val ARBITRARY_FILTER = CardStateFilter.DUE
+
+        // Assert that currentCard is not null before calling onSelectedTags
+        assertNotNull("currentCard should not be null", viewer.currentCard)
+
```

Comment:
perhaps verify all the expected state possible prior to the first action?
(note also that I have a slight preference for positively asserting the exact thing you want, vs negatively asserting one thing from a set of perhaps more than one thing you don't want... thus assert true on question vs assert false on answer - that said I have no idea if that method exists I am just typing this into a text box on github PR reviewer UI)

```suggestion
        assertTrue(viewer.isDisplayingQuestion)
```

Reply:
The method doesn't exist, but still I get what you mean. But i think here it should be fine. But if you think otherwise that is fine too, just hit me up and ill Change it.

Reply:
We typically use: `!viewer.isDisplayingAnswer`

Reply:
got it